caiman_napari/algorithms/cnmfe.py

```diff
@@ -0,0 +1,214 @@
+        downsample_ratio = params['downsample_ratio']
+        # in fname new load in memmap order C
+
+        cn_filter, pnr = cm.summary_images.correlation_pnr(
+            images[::downsample_ratio], swap_dim=False, gSig=gSig
+        )
+
+        if not params['do_cnmfe']:
+             pnr_output_path = str(pathlib.Path(batch_path).parent.joinpath(f"{uuid}_pnr.pikl").resolve())
+             cn_output_path = str(pathlib.Path(batch_path).parent.joinpath(f"{uuid}_cn_filter.pikl").resolve())
+
+             pickle.dump(cn_filter, open(pnr_output_path, 'wb'), protocol=4)
+             pickle.dump(pnr, open(cn_output_path, 'wb'), protocol=4)
+
+             output_file_list = \
+             [
+                 pnr_output_path,
+                 cn_output_path
+             ]
+             print(output_file_list)
+
+             d = dict()
+             d.update(
+                 {
+                     "cnmfe_outputs": output_file_list,
+                     "cnmfe_memmap": fname_new,
+                     "success": True,
+                     "traceback": None
+                 }
+             )
+             print('dict for non-cnmfe:', d)
+
+        else:
+            cnmfe_params_dict = \
+                {
+                    "method_init": 'corr_pnr',
+                    "n_processes": n_processes,
+                    "only_init": True,    # for 1p
+                    "center_psf": True,         # for 1p
+                    "normalize_init": False     # for 1p
+                }
+            tot = {**cnmfe_params_dict, **params['cnmfe_kwargs']}
+            cnmfe_params_dict = CNMFParams(params_dict=tot)
+            cnm = cnmf.CNMF(
+                n_processes=n_processes,
+                dview=dview,
+                params=cnmfe_params_dict
+            )
+            print("Performing CNMFE")
+            cnm = cnm.fit(images)
+            print("evaluating components")
+            cnm.estimates.evaluate_components(images, cnm.params, dview=dview)
+
+            output_path = str(pathlib.Path(batch_path).parent.joinpath(f"{uuid}.hdf5").resolve())
+            cnm.save(output_path)
+
+            d = dict()
+            d.update(
+                {
+                    "cnmfe_outputs": output_path,
+                    "cnmfe_memmap": fname_new,
+                    "success": True,
+                    "traceback": None
+                }
+            )
+            print(d)
+
+    except:
+        d = {"success": False, "traceback": traceback.format_exc()}
+    # Add dictionary to output column of series
+    df.loc[df['uuid'] == uuid, 'outputs'] = [d]
+    # save dataframe to disc
+    df.to_pickle(batch_path)
+
+def load_output(viewer, batch_item: pd.Series):
+    print('loading outputs of CNMFE')
+    path = batch_item["outputs"].item()["cnmfe_outputs"]
+    params = batch_item["params"].item()
+    uuid = batch_item['uuid']
+
+    if not params['do_cnmfe']:
+        cn_filter = pd.read_pickle(path[0])
```

Comment:
I would store the paths in a dict instead of a list so that it's clearer, something like:

```
{'cnmfe_outputs': {'cn': '/bah', 'pnr': '/bahhh'}}
```